### PR TITLE
BUG: to_parquet: accept pathlib.Path for path arg (#4936)

### LINKED ISSUE #4936: Dataframe.to_parquet does not work with Path
```
from dask import dataframe as ddf
from pathlib import Path

# Empty DaskDF for the sake of it 
import pandas as pd
pd_df = pd.DataFrame()
my_df = ddf.from_pandas(pd_df, npartitions=2)

my_df.to_parquet(Path('/mnt/tmp/test'))
```
Last line fails with

```pytb
> ---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-8-d291d9cdd052> in <module>
----> 1 df.to_parquet(Path('/mnt/tmp/test'))

~/.cache/pypoetry/virtualenvs/hello-py3.7/lib/python3.7/site-packages/dask/dataframe/core.py in to_parquet(self, path, *args, **kwargs)
   2904         """ See dd.to_parquet docstring for more information """
   2905         from .io import to_parquet
-> 2906         return to_parquet(self, path, *args, **kwargs)
   2907 
   2908     @derived_from(pd.DataFrame)

~/.cache/pypoetry/virtualenvs/hello-py3.7/lib/python3.7/site-packages/dask/dataframe/io/parquet.py in to_parquet(df, path, engine, compression, write_index, append, ignore_divisions, partition_on, storage_options, compute, **kwargs)
   1252                                          storage_options=storage_options)
   1253     # Trim any protocol information from the path before forwarding
-> 1254     path = infer_storage_options(path)['path']
   1255 
   1256     out = write(df, fs, fs_token, path, write_index=write_index, append=append,

~/.cache/pypoetry/virtualenvs/hello-py3.7/lib/python3.7/site-packages/dask/bytes/utils.py in infer_storage_options(urlpath, inherit_storage_options)
     38     """
     39     # Handle Windows paths including disk name in this special case
---> 40     if re.match(r'^[a-zA-Z]:[\\/]', urlpath):
     41         return {'protocol': 'file',
     42                 'path': urlpath}

/usr/local/lib/python3.7/re.py in match(pattern, string, flags)
    171     """Try to apply the pattern at the start of the string, returning
    172     a Match object, or None if no match was found."""
--> 173     return _compile(pattern, flags).match(string)
    174 
    175 def fullmatch(pattern, string, flags=0):

TypeError: expected string or bytes-like object
```

Is it a problem of implementing a PathLike interface? (On a *non-empty* df using `str(Path(...))` would work)

### PR DESCRIPTION
**Pull request by dlovell**
- [ x ] Tests added / passed
- [ x ] Passes `black dask` / `flake8 dask`

Closes #4936

### CODE REVIEW

File: dask/dataframe/io/tests/test_parquet.py [hunk 1]
```diff
@@ -1688,3 +1688,12 @@ def test_passing_parquetfile(tmpdir):
 
     # should pass, because no need to re-read metadata
     dd.read_parquet(pf)
+
+
+def test_pathlib_path(tmpdir):
```
Comment (mrocklin) [inline dask/dataframe/io/tests/test_parquet.py]: I recommend adding the `engine` pytest fixture here

```suggestion
def test_pathlib_path(tmpdir, engine):
```

File: dask/dataframe/io/tests/test_parquet.py [hunk 2]
```diff
@@ -1688,3 +1688,12 @@ def test_passing_parquetfile(tmpdir):
...
+    import pathlib
+
+    path = pathlib.Path(tmpdir)
+    ddf.to_parquet(path)
+    ddf2 = dd.read_parquet(path)
```
Comment (mrocklin) [inline dask/dataframe/io/tests/test_parquet.py]: and here

```suggestion
    ddf2 = dd.read_parquet(path, engine=engine)
```

File: dask/dataframe/io/tests/test_parquet.py [hunk 3]
```diff
@@ -1688,3 +1688,12 @@ def test_passing_parquetfile(tmpdir):
 
     # should pass, because no need to re-read metadata
     dd.read_parquet(pf)
+
+
+def test_pathlib_path(tmpdir):
+    import pathlib
+
+    path = pathlib.Path(tmpdir)
+    ddf.to_parquet(path)
```
Comment (mrocklin) [inline dask/dataframe/io/tests/test_parquet.py]:
```suggestion
    ddf.to_parquet(path, engine=engine)
```

**Comment by rjzamora** [comment]
Thanks @dlovell!  I just added [a very similar test](https://github.com/dask/dask/pull/4995/commits/e586b6ef6cc66c4b8ec44d4e948974cd0df17270) to the refactor.

File: dask/dataframe/io/tests/test_parquet.py (hunk 1 above)
Comment (dlovell) [reply dask/dataframe/io/tests/test_parquet.py]: removed in favor of @rjzamora 's #4995

File: dask/dataframe/io/tests/test_parquet.py (hunk 2 above)
Comment (dlovell) [reply dask/dataframe/io/tests/test_parquet.py]: removed in favor of @rjzamora 's #4995

**Comment by mrocklin** [comment]
@dlovell any objection to closing this in favor of @rjzamora 's addition of your test to his rewrite PR?

**Comment by dlovell** [comment]
@mrocklin , no objection but I don't see where `pathlib.Path` support was added there.  Should I leave that to @rjzamora  or rebase on top of his MR or something else?

**Comment by rjzamora** [comment]
@dlovell I believe `pathlib.Path` support is covered - I added a `test_pathlib_path` test as a check.

**Comment by dlovell** [comment]
@rjzamora fyi, the docs for `core.to_parquet` only list string as a valid input type

**Pull request closed without merging**